### PR TITLE
Fix indentation of CNI helm chart

### LINKED ISSUE #48472: Wrong indentation for istio-cni repair.labelPods clusterrole
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

### Bug Description

The indentation in the istio-cni ClusterRole for the `labelPods` option is wrong. 

https://github.com/istio/istio/blob/a61a751512256f27bc3b1ff9bc9c38119c6e978c/manifests/charts/istio-cni/templates/clusterrole.yaml#L40-L46

Also we need yaml comments here, not helm comment, otherwise the role wont render correctly.

### Version

```Text
istio version 1.18.6, 1.19.5
```

### Additional Information

Fix:

```
diff --git a/manifests/charts/istio-cni/templates/clusterrole.yaml b/manifests/charts/istio-cni/templates/clusterrole.yaml
index 185dfd3d85..e2f530a812 100644
--- a/manifests/charts/istio-cni/templates/clusterrole.yaml
+++ b/manifests/charts/istio-cni/templates/clusterrole.yaml
@@ -33,10 +33,10 @@ rules:
     resources: ["pods"]
     verbs: ["delete"]
 {{- else if .Values.cni.repair.labelPods }}
-- apiGroups: [""]
-  {{/* pods/status is less privileged than the full pod, and either can label. So use the lower pods/status */}}
-  resources: ["pods/status"]
-  verbs: ["patch", "update"]
+  - apiGroups: [""]
+    # pods/status is less privileged than the full pod, and either can label. So use the lower pods/status
+    resources: ["pods/status"]
+    verbs: ["patch", "update"]
 {{- end }}
 {{- end }}
 ---
@@ -53,7 +53,7 @@ metadata:
     operator.istio.io/component: "Cni"
 rules:
 - apiGroups: [""]
-  {{/* pods/status is less privileged than the full pod, and either can label. So use the lower pods/status */}}
+  # pods/status is less privileged than the full pod, and either can label. So use the lower pods/status
   resources: ["pods/status"]
   verbs: ["patch", "update"]
 {{- end }}

```

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes #48472

This does not impact master branch, so only sending to 1.20

### CODE REVIEW

File: manifests/charts/istio-cni/templates/clusterrole.yaml [hunk 1]
```diff
@@ -32,16 +32,16 @@ rules:
...
   - apiGroups: [""]
     resources: ["pods"]
     verbs: ["delete"]
 {{- else if .Values.cni.repair.labelPods }}
-- apiGroups: [""]
-  {{/* pods/status is less privileged than the full pod, and either can label. So use the lower pods/status */}}
-  resources: ["pods/status"]
-  verbs: ["patch", "update"]
+  - apiGroups: [""]
+    {{- /* pods/status is less privileged than the full pod, and either can label. So use the lower pods/status */}}
```
Comment (Christoph-Raab) [inline manifests/charts/istio-cni/templates/clusterrole.yaml:42]: Tested, rendering works now, thx.

**Comment by jacob-delgado** [comment]
There's going to be a 1.18.7 release? Should we include this?

**Review by zirain — APPROVED** [review]

**Comment by hanxiaop** [comment]
> There's going to be a 1.18.7 release? Should we include this?

I think we do.

**Pull request merged**